net/retriever/retriever.go

```diff
@@ -349,7 +350,7 @@ func (m *MessageRetriever) handleMessage(env pb.Envelope, addr string, id *peer.
 	}
 
 	// Dispatch handler
-	_, err := handler(*id, env.Message, true)
+	resp, err := handler(*id, env.Message, true)
 	if err != nil && err == net.OutOfOrderMessage {
```

Comment:
Every single case here depends on `err != nil`. Pull that out of the condition to a separate level:
```
if err != nil {
  if otherConditions {
  }
}
```